diesel/src/pg/expression/extensions/only_dsl.rs

```diff
@@ -0,0 +1,41 @@
+use crate::query_builder::AsQuery;
+use crate::{query_builder::Only, QuerySource};
+
+/// The `only` method
+///
+/// This is only implemented for the Postgres backend.
+/// The `ONLY` clause is used to select only from one table and not any inherited ones.
+///
+/// Calling this function on a table (`mytable.only()`) will result in the SQL `ONLY mytable`.
+/// `mytable.only()` can be used just like any table in diesel since it implements
+/// [Table](crate::Table).
+///
+/// Example:
+///
+/// ```ignore
+/// let n_users_in_main_table = users::table
+///     .only()
+///     .select(count(users::id))
+///     .first::<i64>(connection)
+/// ```
+/// Selects the number of entries in the `users` table excluding any rows found in inherited
+/// tables.
+///
+/// It can also be used in inner joins:
+///
+/// ```ignore
+/// users::table
+///     .inner_join(posts::table.only())
+///     .select((users::name, posts::title))
+///     .load(connection)
+/// ```
+/// That query excludes any posts that reside in any inherited table.
+///
+pub trait OnlyDsl: AsQuery + Sized + QuerySource + Clone {
+    /// See the trait-level docs.
+    fn only(self) -> Only<Self> {
+        Only { source: self }
+    }
+}
+
+impl<T: QuerySource + AsQuery + Sized + Clone> OnlyDsl for T {}
```

Comment:
Shouldn't that be bound to `Table` only, instead of query source? I mean calling `.only()` on for example an alias is not meaningful, right?